ANTLR/BaseVisitor.py

```diff
@@ -54,10 +59,12 @@ def visitParse(self, ctx: GoPoParser.ParseContext):
 
     def visitAssignment(self, ctx: GoPoParser.AssignmentContext):
         variable_name = ctx.getChild(0).accept(self)
-        # print([method_name for method_name in dir(ctx.getChild(2))
-        #                   if callable(getattr(ctx.getChild(2), method_name))])
         variable_value = ctx.getChild(2).accept(self)
-        self.memory[variable_name] = variable_value
+
+        if self.is_tmp_variable:
+            self.tmp_memory[variable_name] = variable_value
+        else:
+            self.memory[variable_name] = variable_value
```

Comment:
What about example where:

x is defined in memory
is_tmp_variable = true (we are in foreach)
we are doing sth with x

ex.
```
x = 1
list(1...10).foreach(i -> x= x + i)
```

it should be saved in memory and I'm not sure if it will work right now